.github/workflows/dotnet-ci.yml

```diff
@@ -56,3 +56,36 @@ jobs:
           dotnet build --no-restore
           dotnet test --no-build --verbosity normal
           .\.sonar\scanner\dotnet-sonarscanner end /d:sonar.token="${{ secrets.SONAR_TOKEN }}"
+  docker-build:
```

Comment:
We should not try to build image and push to repository every time a pull request is created at the moment as it would waste GitHub action times.

We should have a dedicated pipeline/github action for publish docker image when a branch is updated(For now it is "main" branch).

Let's have a new workflow, that is triggered when "main" receive new commits.

Reply:
I have already set a condition if the current branch is main then push to the registry...  and  I will design another workflow for our projects in a new issue. For this issue, I bring dockerfile and pipeline can build for each PR and only push image when meet the main branch